go/websocketbridge/websocket_bridge_test.go

```diff
@@ -0,0 +1,233 @@
+			marshaledMessage, err := json.Marshal(message)
+			if err != nil {
+				return err
+			}
+			_, err = clientWebsocket.Write(marshaledMessage)
+			if err != nil {
+				return err
+			}
+		}
+		return err
+	})
+
+	s.Step(`receiving this message from exocom:`, func(payloadStr *gherkin.DocString) error {
+		var message structs.Message
+		err := json.Unmarshal([]byte(payloadStr.Content), &message)
+		if err != nil {
+			return err
+		}
+		err = exocom.WaitForConnection()
+		if err != nil {
+			return err
+		}
+		if message.SessionID != "" {
+			for key := range websocketBridgeInstance.SocketMap {
+				message.SessionID = key
+			}
+		}
+		return exocom.Send(message)
+	})
+
+	s.Step(`^those clients have different sessionIds$`, func() error {
+		if len(websocketBridgeInstance.SocketMap) != clientCount {
+			return fmt.Errorf("Expected request to have %d clients but got %d", clientCount, len(websocketBridgeInstance.SocketMap))
+		}
+		return nil
+	})
+
+	s.Step(`websocket bridge makes the websocket request:`, func(payloadStr *gherkin.DocString) error {
+		var expectedMessage structs.Message
+		err := json.Unmarshal([]byte(payloadStr.Content), &expectedMessage)
+		if err != nil {
+			return err
+		}
+		actualMessage, err := exocom.WaitForMessageWithName(expectedMessage.Name)
+		if err != nil {
+			return err
+		}
+		//ID and SessionId get generated on the fly, so we can't know them beforehand
+		expectedMessage.ID = actualMessage.ID
+		expectedMessage.SessionID = actualMessage.SessionID
+		if !reflect.DeepEqual(actualMessage, expectedMessage) {
+			return fmt.Errorf("Expected request to equal %s but got %s", expectedMessage, actualMessage)
+		}
+		return nil
+	})
+
+	s.Step(`websocket bridge sends this to the client:`, func(payloadStr *gherkin.DocString) error {
+		var bytes []byte
+		err := websocket.Message.Receive(clientWebsocket, &bytes)
+		if err != nil {
+			return err
+		}
+		var actualMessage structs.Message
+		err = json.Unmarshal(bytes, &actualMessage)
+		if err != nil {
+			return err
+		}
+		var expectedMessage structs.Message
+		err = json.Unmarshal([]byte(payloadStr.Content), &expectedMessage)
+		if err != nil {
+			return err
+		}
+		if !reflect.DeepEqual(actualMessage, expectedMessage) {
+			return fmt.Errorf("Expected request to equal %s but got %s", expectedMessage, actualMessage)
+		}
+		return nil
+	})
+
+	s.Step(`^the websocket bridge does not send a message to the client$`, func() error {
+		var bytes []byte
+		clientWebsocket.SetReadDeadline(time.Now().Add(time.Second * time.Duration(1)))
+		err := websocket.Message.Receive(clientWebsocket, &bytes)
+		if strings.Contains(err.Error(), "i/o timeout") {
+			if bytes != nil {
+				var actualMessage structs.Message
+				err = json.Unmarshal(bytes, &actualMessage)
+				if err != nil {
+					return err
+				}
+				return fmt.Errorf("Expected no message, but received %v", actualMessage)
+			}
+			return nil
+		} else {
+			fmt.Printf(err.Error())
```

Comment:
I don't think we need to print the error here since we are returning it, which would cause godog to print it.